src/NUnitCommon/nunit.agent.core/AgentOptions.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using NUnit.Engine;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace NUnit.Agents
+{
+    /// <summary>
+    /// All agents, either built-in or pluggable, must be able to 
+    /// handle the options defined in this class. In some cases,
+    /// it may be permissible to ignore them but they should never
+    /// give rise to an error.
+    /// </summary>
+    public class AgentOptions
+    {
+        static readonly char[] DELIMS = new[] { '=', ':' };
+        // Dictionary containing valid options with bool value true if a value is required.
+        static readonly Dictionary<string, bool> VALID_OPTIONS = new Dictionary<string, bool>();
+
+        static AgentOptions()
+        {
+            VALID_OPTIONS["agentId"] = true;
+            VALID_OPTIONS["agencyUrl"] = true;
+            VALID_OPTIONS["debug-agent"] = false;
+            VALID_OPTIONS["debug-tests"] = false;
+            VALID_OPTIONS["trace"] = true;
+            VALID_OPTIONS["pid"] = true;
+            VALID_OPTIONS["work"] = true;
+        }
+
+        public AgentOptions(params string[] args)
+        {
+            int index;
+            for (index = 0; index < args.Length; index++)
+            {
+                string arg = args[index];
+
+                if (IsOption(arg))
+                {
+                    var option = arg.Substring(2);
+                    var delim = option.IndexOfAny(DELIMS);
+                    var opt = option;
+                    string? val = null;
+                    if (delim > 0)
+                    {
+                        opt = option.Substring(0, delim);
+                        val = option.Substring(delim + 1);
+                    }
+
+                    if (!VALID_OPTIONS.ContainsKey(opt))
```

Comment:
nitpick: Could this be simplified to something like the below to save the duplicate dictionary lookup?
```
if (!VALID_OPTIONS.TryGetValue(opt, out bool optionTakesValue))
   throw new Exception($"Invalid argument: {arg}");
```

Reply:
Yes. The code will be less clear, I think, bu tI can add a comment.